File: DSA/DSA05012 TỔ HỢP C(n, k).cpp
#include <bits/stdc++.h>
using namespace std;

const int MOD = 1e9 + 7;

vector<vector<long long>> prepare()
{
    vector<vector<long long>> save(1001, vector<long long>(1001, 1));

    for (int i = 2; i < 1001; i++)
    {
        for (int j = 1; j < i; j++)
        {
            if (j == 1) save[i][j] = i;
            else save[i][j] = (save[i-1][j] + save[i-1][j-1])%MOD;
        }
    }
    return save;
}

void submit(vector<vector<long long>> &save)
{
    int n, k;
    cin >> n >> k;

    
    cout << save[n][k]%MOD;
    
    
    cout << endl;
}



int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);
    
    int so_bo_test;
    cin >> so_bo_test;
    cin.ignore();
    vector<vector<long long>> save = prepare();
    
    while (so_bo_test--)
    {
        submit(save);
    }

    return 0;
}

File: DSA_python/DSA05012 TỔ HỢP C(n, k).py
def prepare():
    save = [[0]*1001 for _ in range(1001)]
    for i in range(1001):
        save[i][0] = save[i][i] = 1
        for j in range(1, i):
            save[i][j] = (save[i-1][j]+save[i-1][j-1])%MOD
    return save

def submit():
    n, k = map(int, input().split())

    print(save[n][k]%MOD)

test = int(input())
MOD = 10**9 +7
save = prepare()
for _ in range(test):
    submit()
